src/target/riscv/riscv.c

```diff
@@ -462,11 +462,76 @@ static void trigger_from_breakpoint(struct trigger *trigger,
 	trigger->unique_id = breakpoint->unique_id;
 }
 
-static int maybe_add_trigger_t1(struct target *target,
-		struct trigger *trigger, uint64_t tdata1)
+static bool match_napot_checker(struct trigger *trigger, riscv_reg_t *tdata2)
+{
+	riscv_reg_t addr = trigger->address;
+	riscv_reg_t size = trigger->length;
+	if (size > 1 &&
+		(size & (size - 1)) == 0 &&
+		(addr & (size - 1)) == 0) {
```

Comment:
If you'd like, you can further improve the readability:

```suggestion
	bool sizePowerOf2 = (size & (size - 1)) == 0;
	bool addrAligned = (addr & (size - 1)) == 0;	
	if (size > 1 && sizePowerOf2 && addrAligned) {
```